docs/localization/LOCALIZATION.md

```diff
@@ -216,6 +216,66 @@ As a result "**hello bold**" will be printed inside the span element.
 
 ```
 
+### How to generate localization files for a component
+
+You can directly generate a localized component with the following command:
+```shell
+ng g component ComponentName --use-localization
+```
+Or you can localized it later with:
+```shell
+ng g @o3r/localization:add-localization --path="/path/for/class.component.ts"
+```
+
+> [!NOTE]
+> Later you can add localization key with the following command:
+> ```shell
+> ng g @o3r/localization:add-localization-key --path="/path/for/class.component.ts" --key=myNewLocalizationKey
+> ```
+
+#### Generated files
+
+##### Translation (`*.translation.ts`)
+
+It is used to define the localization variables used by component template. my-component.translation.ts file typically defines an interface which extends `Translation` from `@o3r/core` with all possible variable names used by your presenter template. It also exports a constant which satisfies the above contract. The values for each property are localization keys (real keys from localization bundle).
+
+```typescript
+import {Translation} from  '@o3r/core';
+
+export  interface MyComponentPresTranslation extends Translation {
+  prop1: string;
+  prop2: string;
+}
+
+export  const translations: MyComponentPresTranslation = {
+  prop1: 'o3r-my-component-pres.somekey1',
+  prop2: 'o3r-my-component-pres.somekey2'
+};
+```
+
+##### Localization (`*.localization.json`)
+
+It defines an object being key/value pairs. Each value is a json object having `description` and `defaultValue` properties. Eventually you can reference a global key via $ref using relative path to `global-localization.json` which sits in `src` or in different package in dependencies. The purpose of this file is to provide a default localization for component so that library user can start building pages using components without worrying about localization. `*.localization.json` specifies only default values in english.
```

Comment:
```suggestion
It defines an object of key/value pairs. Each value is a JSON object having `description` and `defaultValue` properties. Eventually you can reference a global key via `$ref` using a relative path to `global-localization.json` which sits in `src` or in different package in dependencies. The purpose of this file is to provide a default localization for a component so that library user can start building pages using components without worrying about localization. The `*.localization.json` file specifies default values only in English.
```